### ISSUE TITLE
Cannot disable distributed optimizer for single-GPU unified memory systems

### ISSUE THREAD

**Issue by AmitMY** [NVIDIA-NeMo/Megatron-Bridge#2302]
## Problem

Setting `cfg.ddp.use_distributed_optimizer = False` is silently overridden back to `True` with a warning:

```
Distributed optimizer settings were not in sync: ddp.use_distributed_optimizer=False, 
optimizer.use_distributed_optimizer=True. Automatically enabling distributed optimizer for both settings.
```

Setting both `cfg.ddp.use_distributed_optimizer = False` and `cfg.optimizer.use_distributed_optimizer = False` may work, but the auto-sync behavior makes it unclear whether this is a supported configuration.

## Why this matters

On **unified memory platforms** (e.g. NVIDIA GB10 / DGX Spark, where CPU and GPU share 120GB), the distributed optimizer offloads fp32 optimizer states to CPU RAM — but CPU and GPU draw from the same memory pool. With a single GPU there's no sharding benefit, so the distributed optimizer only adds CPU↔GPU data movement overhead without saving any memory.

In our setup (60M model, MBS=26, SEQ_LENGTH=4096), the training process uses:
- **77 GB** GPU-side (model + activations + gradients)
- **34 GB** CPU-side (optimizer states + data workers)
- **Total: ~111 GB** out of 120 GB shared pool

Disabling the distributed optimizer would consolidate optimizer states on GPU, eliminating the data transfer overhead while using the same total memory.

## Environment

- **NeMo container**: `nvcr.io/nvidia/nemo:25.11`
- **GPU**: NVIDIA GB10 (DGX Spark), compute capability 12.1, 120 GB unified memory
- **Training**: single-GPU pretraining via `megatron.bridge.training.pretrain`
- **Recipe**: `qwen3_600m_pretrain_config()`

## Expected behavior

`cfg.ddp.use_distributed_optimizer = False` should be respected (or there should be a single config knob that controls both DDP and optimizer settings consistently).

**Comment by AmitMY** [comment]
Setting both `cfg.ddp.use_distributed_optimizer = False` and `cfg.optimizer.use_distributed_optimizer = False` works — no override warning, training runs normally.

The issue is that setting only `cfg.ddp.use_distributed_optimizer = False` triggers auto-sync that re-enables it. The fix for users is to set both flags. 

Suggestion: either make this a single config knob, or at least document that both must be set together. The current auto-sync behavior silently overrides the user's intent.

**Comment by yaoyu-33** [comment]
Thanks for suggestion.
Internally we see reporting of the issue as well. We will put all auto sync stuff together in the code and make clearer loggings.

**Comment by shaltielshmid** [comment]
I encountered this as well - initially I was OOM'ing on smaller workloads until I encountered this issue. Perhaps it is worth defaulting to False when working on unified memory/single GPU?

Thanks @AmitMY, good find.

**Comment by yaoyu-33** [comment]
Closing as part of issue cleanup — this issue has been stale with no recent activity. If you still need help, please open a new issue with updated details. Thank you!
